libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -1479,7 +1486,221 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_CancelFuzzing )
 }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Assumes HW loopback where MOSI --> MISO.
+ *        Verify the happy-path of a syncronous read
+ *
+ */
+TEST( TEST_IOT_SPI, IotSPI_LoopBack_ReadSync )
+{
+    IotSPIHandle_t xSPIHandle;
+    int32_t lRetVal;
+    IotSPIMasterConfig_t xConfig;
+    const uint32_t ulNBytes = 16;
+    uint8_t pucRxBuf[ ulNBytes ];
+
+    memset( pucRxBuf, 0, ulNBytes );
+
+    xSPIHandle = iot_spi_open( ultestIotSpiInstance );
+    TEST_ASSERT_NOT_EQUAL( NULL, xSPIHandle );
+
+    if( TEST_PROTECT() )
+    {
+        /* Adjust the dummy value */
+        TEST_ASSERT( IOT_SPI_SUCCESS == iot_spi_ioctl( xSPIHandle, eSPIGetMasterConfig, &xConfig ) );
+        xConfig.ucDummyValue = 0xDC;
+        TEST_ASSERT( IOT_SPI_SUCCESS == iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xConfig ) );
+
+        /* Install callback which should not be exercised because it's sync */
+        iot_spi_set_callback( xSPIHandle, prvSpiAsyncCallback, NULL );
+
+        /* The callback should not be called to give sem back. Read, verify callback not called */
+        TEST_ASSERT( IOT_SPI_SUCCESS == iot_spi_read_sync( xSPIHandle, pucRxBuf, ulNBytes ) );
+        TEST_ASSERT( pdFAIL == xSemaphoreTake( xtestIotSPISemaphore, testIotSPI_DEFAULT_SEMAPHORE_DELAY / 2 ) );
+
+        /* Verify expected bytes. Because it's loopback, the configured dummy value
+         * should be transmitted, and thus read, for all N queried bytes.*/
+        TEST_ASSERT_EACH_EQUAL_UINT8( xConfig.ucDummyValue, pucRxBuf, ulNBytes );
+    }
+
+    lRetVal = iot_spi_close( xSPIHandle );
```

Comment:
I think the implementation should make sure the dummy value is reset to 0 in the close function.

Reply:
Could you elaborate on this? Do you mean restoring the configuration to default values so the next open will also have initial values? Or just the dummy value?

Reply:
Sorry, I confused you. I mean restoring the configuration to default values in next open. Dummy value is just one of the them.

The restore can happen in close or open.

Reply:
I like this. Adding this now.